doc/Variables.md

```diff
@@ -0,0 +1,144 @@
+# Variables
+
+In the fuse stack, a Variable does two things:
+
+## Variable Value
+
+First, a Variable has a _value_. A Variable holds one or more dimensions of floating point data that represents some
+physical or logical property. For example, a two dimensional pose consists of `x`, `y`, and `yaw` floating point
+values. A single Variable class could be implemented hold all three pose dimensions. The exact value of each dimension
+will be unknown and uncertain. It is the job of an Optimizer to find the most likely value of each dimension held by
+each Variable instance in the system.
+
+## Variable Identity
+
+Second, a Variable must have an _identity_, some way of distinguishing one occurrence of a Variable from another within
+the system. There are several reasons multiple identities of a Variable will be contained within a single system:
+
+* The Variable may represent a property common to multiple entity types. Both a robot and a visual landmark have a
+  position in space. Different identities of the same Variable type may be used to describe these different entities.
+* Similarly, there may be multiple occurrences of the same entity type within the system. A multi-robot configuration
+  will want to track the pose of each robot, and thus different identities of a Variable will be used for each robot
+  in the system.
+* Most commonly, a Variable will represent a time-varying process. A different Variable identity will be required
+  for each time instant for which the process value is to be estimated. For example, the pose of the robot will change
+  over time, so we need a unique Variable identity representing the robot pose at time `t1` **and** another unique
+  Variable identity representing the robot pose at time `t2`. Any time-varying process must be discretized within
+  the fuse stack.
+
+The Variable identity takes the form of a UUID or hash value, and is generally derived from a set of additional
+properties that describe what makes each occurrence unique from other occurrences. In the case of a time-varying
+process, this will likely involve the timestamp. In the case of a Variable that can describe multiple robots, this
+might involve the serial number of the robot.
+
```

Comment:
Tricky and time-consuming, but some diagrams to illustrate each of the uses of the identity might be an aid to more visual learners. Even just something linking the component parts of the identity to bot positions, a time series, etc.